### DISCUSSION TITLE
How To Plot the Satellite Attitude Modes

### DISCUSSION THREAD

**Discussion by chilk-gnss** [groops-devs/groops#188] [Q&A]
Dear Developers,

If a solar panel obstructs the antenna's field of view in certain directions, it can cause a reduction in signal strength or introduce additional phase variations. This can result in a shift in the apparent phase center, impacting the accuracy of the antenna's measurement. 

I am trying to relate the shape of the satellite, signal strength, and phase variations in the simulation and calibration of satellites. I was trying to learn how I can model the shadow for a given constellation such as Galileo or BDS. I have gone through the [SimulateStarCameraGnss](https://groops-devs.github.io/groops/html/SimulateStarCameraGnss.html) and the [GnssAttitudeInfoCreate](https://groops-devs.github.io/groops/html/GnssAttitudeInfoCreate.html) programs. For example, I have some files given below:

1. [attitude_2022-04-20.C37.txt](https://github.com/groops-devs/groops/files/11480091/attitude_2022-04-20.C37.txt)
2. [orbit_2022-04-20.C37.txt](https://github.com/groops-devs/groops/files/11480111/orbit_2022-04-20.C37.txt)

As a starting point, could you show me how I can plot the attitude modes used by a single system as indicated above? Here is the file for BDS-2I/2M/3I attitude modes. 

- [bdsAttitude.txt](https://github.com/groops-devs/groops/files/11480192/bdsAttitude.txt)

How can I show the elevation of the sun with respect to the orbital place for a particular satellite on a certain day(for example as shown [here)?](https://groops-devs.github.io/groops/html/GnssAttitudeInfoCreate.html)

Thanks in advance.

Chilk.

**Comment by tmayerguerr** [comment]
Dear Chilk,

I don't know the full answer to your question. But I can give you a few ideas:

- With [Orbit2EclipseFactor](https://groops-devs.github.io/groops/html/Orbit2EclipseFactor.html) you can determine the shadow periods.
- [Orbit2BetaPrimeAngle](https://groops-devs.github.io/groops/html/Orbit2BetaPrimeAngle.html) computes the beta prime angle (between the orbital plane and earth-sun direction)
- You can use [SimulateStarCameraGnss](https://groops-devs.github.io/groops/html/SimulateStarCameraGnss.html) without inputfileAttitudeInfo to generate nominal yaw-steering attitude.
-  Use [InstrumentStarCameraMultiply](https://groops-devs.github.io/groops/html/InstrumentStarCameraMultiply.html) with your real and nominal attitude (with one in inverse direction) to compute the difference rotation.
- The yaw angle is computed with [InstrumentStarCamera2RollPitchYaw](https://groops-devs.github.io/groops/html/InstrumentStarCamera2RollPitchYaw.html)

Hopes this helps you to find a solution
regards
Torsten

**Comment by chilk-gnss** [comment]
Dear Professor @tmayerguerr,

I sincerely appreciate the time you dedicated to answering my questions, explaining intricate concepts, and providing hands-on assistance whenever needed. Using the advice, I managed to get the yaw:

For example ==

- [yaw_2023-05-01.C32.txt](https://github.com/groops-devs/groops/files/11589478/yaw_2023-05-01.C32.txt)


I tried to compare the solutions from different attitude modes. For example, these are the residuals for model A and model B. 

1. [timeSeriesB_2023-05-01.hofn.txt](https://github.com/groops-devs/groops/files/11589454/timeSeriesB_2023-05-01.hofn.txt)
2. [timeSeriesA_2023-05-01.hofn.txt](https://github.com/groops-devs/groops/files/11589455/timeSeriesA_2023-05-01.hofn.txt)

When I plot them, the residuals perfectly match even in the eclipse. Is that situation OK using standard attitude vs .obx? In the case of replicating, the results are for 

**- station:** dgar
**- prn:** 32
**- system:** C
**- phase residual:** band 5

Am not sure, has the continuous Yaw-Steering have anything to do with this? 

Thanks in advance.

Regards,

Chilk.

**Comment by gnssLa** [comment]
@tmayerguerr,

I was following your advice on the issue above. 

I am just curious. How can I estimate the **Phase Center Variations** of a particular PRN with respect to **Nadir Angle** in GROOPS? 

**NB:** given any two frequencies. 

Thanks in advance for your assistance.

Regards,

Duarte.
